macos/FluentUI/Core/Colors.swift

```diff
@@ -438,22 +438,115 @@ public final class Colors: NSObject {
+		get {
+			if let colorProvider = colorProvider {
+				return colorProvider.primary
+			} else {
+				return _primary
+			}
+		}
+		set {
+			_primary = newValue
+		}
+	}
+	@objc public static var primaryShade10: NSColor {
+		get {
+			if let colorProvider = colorProvider {
+				return colorProvider.primaryShade10
+			} else {
+				return _primaryShade10
+			}
+		}
+		set {
+			_primaryShade10 = newValue
+		}
+	}
+	@objc public static var primaryShade20: NSColor {
+		get {
+			if let colorProvider = colorProvider {
+				return colorProvider.primaryShade20
+			} else {
+				return _primaryShade20
+			}
+		}
+		set {
+			_primaryShade20 = newValue
+		}
+	}
+	@objc public static var primaryShade30: NSColor {
+		get {
+			if let colorProvider = colorProvider {
+				return colorProvider.primaryShade30
+			} else {
+				return _primaryShade30
+			}
+		}
+		set {
+			_primaryShade30 = newValue
+		}
+	}
+	@objc public static var primaryTint10: NSColor {
+		get {
+			if let colorProvider = colorProvider {
+				return colorProvider.primaryTint10
+			} else {
+				return _primaryTint10
+			}
+		}
+		set {
+			_primaryTint10 = newValue
+		}
+	}
+	@objc public static var primaryTint20: NSColor {
+		get {
+			if let colorProvider = colorProvider {
+				return colorProvider.primaryTint20
+			} else {
+				return _primaryTint20
+			}
+		}
+		set {
+			_primaryTint20 = newValue
+		}
+	}
+	@objc public static var primaryTint30: NSColor {
+		get {
+			if let colorProvider = colorProvider {
+				return colorProvider.primaryTint30
+			} else {
+				return _primaryTint30
+			}
+		}
+		set {
+			_primaryTint30 = newValue
+		}
+	}
+	@objc public static var primaryTint40: NSColor {
+		get {
+			if let colorProvider = colorProvider {
+				return colorProvider.primaryTint40
+			} else {
+				return _primaryTint40
+			}
+		}
+		set {
+			_primaryTint40 = newValue
+		}
+	}
 
-	@objc public static var primary: NSColor = Colors.Palette.communicationBlue.color
-
-	@objc public static var primaryTint10: NSColor = Colors.Palette.communicationBlueTint10.color
-
-	@objc public static var primaryTint20: NSColor = Colors.Palette.communicationBlueTint20.color
-
-	@objc public static var primaryTint30: NSColor = Colors.Palette.communicationBlueTint30.color
-
-	@objc public static var primaryTint40: NSColor = Colors.Palette.communicationBlueTint40.color
-
-	@objc public static var primaryShade10: NSColor = Colors.Palette.communicationBlueShade10.color
-
-	@objc public static var primaryShade20: NSColor = Colors.Palette.communicationBlueShade20.color
+	/// Color provider object. When a Color Provider is present, colors will be retrieved from the provider vs. internally.
+	@objc public static var colorProvider: ColorProviding?
 
-	@objc public static var primaryShade30: NSColor = Colors.Palette.communicationBlueShade30.color
+	/// These are initialized with the Communication Blue defaults.
+	private static var _primary = Colors.Palette.communicationBlue.color
```

Comment:
Actually do we need to have both private and public vars for these color values? Maybe we could just have public values, initialize them with default comm blue and have a setter in colorProvider that updates them to the passed in values in the provider?

Reply:
If I'm understanding correctly, the problem is that would then ruin the benefit of the provider model - lazy loading of colors when we actually use them. Because then we'd have to load every color when the colorProvider is set.

Reply:
That was my original intention, but this class isn't instantiated so Swift doesn't allow that. I can use static var's but it won't allow an assigned value and a get - I get errors like "Cannot find 'get' in scope.

Reply:
I don't think the provider model does lazy loading of branding colors on the iOS side though. There is no `didSet` but we have a `setColorProvider` function that essentially works like a didSet and overrides all the branding colors as soon as a provider is set.

Reply:
Ooh that seems like a potential improvement we can make to iOS since it kinda defeats the purpose of the provider pattern?